src/main/java/org/openRealmOfStars/AI/Mission/MissionHandling.java

```diff
@@ -1965,40 +1965,41 @@ public static void findScoutShipForMission(final PlayerInfo info,
       Fleet fleet = info.getFleets().getByIndex(j);
       Mission fleetMission = info.getMissions().getMissionForFleet(
           fleet.getName());
+
       if (fleetMission == null
           || mission.getType() == MissionType.DIPLOMATIC_DELEGACY
-             && fleetMission.getType() == MissionType.MOVE
-             && fleetMission.getMissionTime() > 4) {
-        for (Ship ship : fleet.getShips()) {
-          Fleet newFleet = null;
-          if (ship.isScoutShip()) {
-            if (fleet.getNumberOfShip() > 1) {
-              fleet.removeShip(ship);
-              newFleet = new Fleet(ship, fleet.getX(), fleet.getY());
-            } else {
-              newFleet = fleet;
-              if (fleetMission != null
-                  && mission.getType() == MissionType.DIPLOMATIC_DELEGACY
-                  && fleetMission.getType() == MissionType.MOVE) {
-                info.getMissions().remove(fleetMission);
-              }
-            }
-          }
-          if (newFleet != null) {
-            if (mission.getType() == MissionType.EXPLORE) {
-              newFleet.setName(info.getFleets()
-                  .generateUniqueName("Scout"));
-              mission.setFleetName(newFleet.getName());
-              mission.setPhase(MissionPhase.LOADING);
-            }
-            if (mission.getType() == MissionType.DIPLOMATIC_DELEGACY) {
-              newFleet.setName(info.getFleets()
-                  .generateUniqueName(newFleet.getFirstShip().getName()));
-              mission.setFleetName(newFleet.getName());
-              mission.setPhase(MissionPhase.LOADING);
-            }
+              && fleetMission.getType() == MissionType.MOVE
+              && fleetMission.getMissionTime() > 4) {
+        return;
+      }
+
+      for (Ship ship : fleet.getShips()) {
+        Fleet newFleet = fleet.split(ship);
```

Comment:
This would always split the fleet or if there is a single ship it would then return that just as newFleet. Old code checks if fleet has scout ship and it removes it to make a new fleet. In old code there is also a but it may assign this new mission to all scout ships in same fleet which is not intentional.